src/CoreShop/Payum/PostFinance/EventListener/TokenInvalidationListener.php

```diff
@@ -8,53 +8,24 @@
  *
  * @copyright  Copyright (c) 2015-2020 Dominik Pfaffenbauer (https://www.pfaffenbauer.at)
  * @license    https://www.coreshop.org/license     GNU General Public License version 3 (GPLv3)
-*/
+ */
 
 namespace CoreShop\Payum\PostFinanceBundle\EventListener;
 
 use CoreShop\Component\Core\Configuration\ConfigurationServiceInterface;
 use CoreShop\Payum\PostFinanceBundle\Invalidator\TokenInvalidatorInterface;
-use Pimcore\Event\System\MaintenanceEvent;
-use Pimcore\Model\Schedule\Maintenance\Job;
+use Pimcore\Maintenance\TaskInterface;
 
-final class TokenInvalidationListener
+final class TokenInvalidationListener implements TaskInterface
 {
-    /**
-     * @var ConfigurationServiceInterface
-     */
-    private $configurationService;
-
-    /**
-     * @var TokenInvalidatorInterface
-     */
-    private $tokenInvalidator;
-
-    /**
-     * @var int
-     */
-    private $days;
-
-    /**
-     * TokenInvalidationListener constructor.
-     *
-     * @param ConfigurationServiceInterface $configurationService
-     * @param TokenInvalidatorInterface     $tokenInvalidator
-     * @param int                           $days
-     */
     public function __construct(
-        ConfigurationServiceInterface $configurationService,
-        TokenInvalidatorInterface $tokenInvalidator,
-        $days = 0
+        private readonly ConfigurationServiceInterface $configurationService,
```

Comment:
readonly types are PHP 8.1 only, CoreShop and Pimcore have 8.0 as min requirement